Entities/Vehicles/Common/VehicleCommon.as

```diff
@@ -551,15 +547,36 @@ void Vehicle_StandardControls(CBlob@ this, VehicleInfo@ v)
 		for (uint i = 0; i < aps.length; i++)
 		{
 			AttachmentPoint@ ap = aps[i];
-			CBlob@ blob = ap.getOccupied();
+			CBlob@ occupied = ap.getOccupied();
+			
+			//LET GO
+			AttachmentPoint@ pickup = this.getAttachments().getAttachmentPointByName("PICKUP");
+			if (pickup !is null)
+			{
+				CBlob@ carrier = pickup.getOccupied();
+				
+				if (carrier !is null)
+				{
+					if (this.getTeamNum() != carrier.getTeamNum())
+					{
+						CBitStream params;
+						params.write_u16(carrier.getNetworkID());
+						this.SendCommand(this.getCommandID("vehicle letgo"), params);
+						return;
+					}
+				}
+			} // let go
 
-			if (blob !is null && ap.socket)
+			if (occupied !is null && ap.socket)
 			{
 				// GET OUT
-				if (blob.isMyPlayer() && ap.isKeyJustPressed(key_up))
-				{
+				if 	(occupied.isMyPlayer() && ap.isKeyJustPressed(key_up)
+					|| (this.getTeamNum() != occupied.getTeamNum() 
+						&& (this.getTeamNum() >= 0 && this.getTeamNum() <= 7)
+						&& (occupied.getName() == "knight" || occupied.getName() == "archer" || occupied.getName() == "builder")))
```

Comment:
"knight", "archer" and "builder is hard coded in a lot more places. 
The ItemLimits script I made will also check for team number like this.
How to avoid this?
Although neutral blobs are usually set to -1 or 255, you can change to teams 8~254 via `!team (number)` in Sandbox. Is there a way to check if a blob is in grey team that doesn't involve checking team number?

Reply:
> "knight", "archer" and "builder is hard coded in a lot more places.
> How to avoid this?

You should always endeavour to leave code in a better state than you found it. A potential solution would be to check for `"player"` tag, but migrants also have this tag which could cause rare side effects.

> Although neutral blobs are usually set to -1 or 255, you can change to teams 8~254 via !team (number) in Sandbox. Is there a way to check if a blob is in grey team that doesn't involve checking team number?

There is nothing special with grey teams. They are only grey because there aren't enough colours to cover all team numbers. The only team that should be treated differently from the rest is spectator team (`getRules().getSpectatorTeamNum()`). -1 == 255 because team is u8 which wraps negative numbers around to the maximum.

Reply:
I can change to `occupied.hasTag("player")`. I don't see a problem since migrants are never attached to vehicles.

With the team check, this kind of check is in many places. I guess teams 0~254 should be treated as player teams and team 255 as a neutral team for animals and stuff that can be used by anyone (vehicles, doors).
I have seen places where the team check says "anything that is higher than team 100" or "anything that is higher than team 10". Those checks seem arbitrary and inconsistent. Up until now, I tried to be a more precise by handling teams 0~7 as player teams and anything else as neutral team.

I can change that so that 0~254 are player teams instead. And 255 is a neutral team.
Is that what should be done?

Reply:
> I can change that so that 0-254 are player teams instead. And 255 is a neutral team. Is that what should be done?

Yes, this sounds like the most appropriate way of dealing with teams. Please remember to check for `getRules().getSpectatorTeamNum()` rather than specifically team 255.

Reply:
Sorry if it was unclear. I was not talking about spectator team.
I'm still talking about **teams that are currently on the field playing**. Of those, 0 to 254 should be handled as teams that players could get assigned to. Players can also get assigned to team 255, but behavior will be different.

Reply:
Hold up... why is there even a team or player check here? If the blob is already sitting on the vehicle, they must've been the correct team to get on it in the first place. And there is no need to check if it's a player since non-player blobs will never be pressing `key_up` to get out. Even if they somehow did, there is no harm in allowing them to detach from the vehicle like players do.

Looking at SeatHop.as line 32, there already appears to be a check to allow neutral teams to sit in vehicles.

Reply:
It is so that when you are sitting in your vehicle and it is getting converted to the enemy, you are kicked out.
The player check is so that enemy crates and enemy Dinghies can be loaded/can stay in the catapult's mag.

I'm not working on "neutral players doing stuff" but I'm working on "any players using neutral stuff".

About `getRules().getSpectatorTeamNum()`, I can use it, but the Map loader will set blobs to -1 so I'm afraid of mismatch errors.

Reply:
Team is a `u8` which means -1 wraps around to 255 (the maximum). `getRules().getSpectatorTeamNum()` returns 255 so it should match correctly.

Reply:
In Sandbox mode, `getRules().getSpectatorTeamNum()` returns 200 for me.
I'm reworking team checks and adding another commit later.